src/control/server/server_utils_test.go

```diff
@@ -469,8 +469,13 @@ func TestServer_prepBdevStorage(t *testing.T) {
 				cfg = tc.srvCfgExtra(cfg)
 			}
 
+			// ensure that the engine affinities are set.
+			if err := cfg.SetEngineAffinities(log); err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Minor, but I suggest setting the affinities manually in this test instead of calling SetEngineAffinities. The SetEngineAffinities logic is nontrivial, and could complicate debugging the test.

Reply:
Agreed. That having been said, I think the `Validate()` method(s) still have too many side-effects and should be refactored to make them read-only validations. This test relies on those side-effects and should be updated to be more explicit about the inputs. I don't think that necessarily needs to be part of this PR, though.

Reply:
Agreed on all counts.